python/pycrdt/_awareness.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+import json
+import time
+from typing import Any, Callable
+from uuid import uuid4
+
+from ._doc import Doc
+from ._sync import Decoder, YMessageType, read_message, write_var_uint
+
+DEFAULT_USER = {"username": str(uuid4()), "name": "Jupyter server"}
```

Comment:
Should we have a static UUID, or should it be dynamically generated like you did?

Reply:
It can remain dynamic, and the user can be set later if a specific username is required.

Reply:
Maybe we should user a proper identity https://jupyter-server.readthedocs.io/en/latest/operators/security.html#identity-providers, but I don't know if we can depend on jupyter-server...

Reply:
No we cannot depend on jupyter-server, and actually having "Jupyter server" as the name is also problematic. Maybe there shouldn't be a default user, and the `Awareness` should always be passed a user?

Reply:
But that would be a breaking change.
I think we should be able to use the awareness without necessarily adding a new user, so if no user is passed, then we don't add it in the states. What do you think?

Reply:
Let do that, even if the state will hardly be readable without user.
Should we keep the `user` then, or treat it as any other state ?

Reply:
Looking at [awareness.js](https://github.com/yjs/y-protocols/blob/255b7e6c96d9c6dd48b5519b3f83ab65afdb4bba/awareness.js), there is nothing about users, so I think we should do the same and remove it.

Reply:
Done in https://github.com/jupyter-server/pycrdt/pull/170/commits/f523e3dc03265af37df0483f5e0716a7c9e44c39